pyoomph/types.py

```diff
@@ -64,7 +64,6 @@ def __repr__(self) -> str:
     def set_type_members(self, type_members: List[Type]) -> None:
         assert len(type_members) >= 2
         assert len(type_members) == len(set(type_members))  # no duplicates
-        assert all(t.refcounted for t in type_members)  # TODO: get rid of this assert
```

Comment:
can this assert really be deleted?

Reply:
It can. Even though the union will be refcounted, increffing and decreffing it won't actually do anything.